packages/react-examples/.storybook/preview.js

```diff
@@ -5,68 +5,69 @@ import 'cypress-storybook/react';
- * @type {string}
- */
-const packageNamePlaceholder = 'PACKAGE_NAME';
-
 initializeIcons();
 
+// This API is deprecated (in favor of `export const decorators = []`) but the new way appears not
+// to work when using the legacy configure() API
 addDecorator(withPerformance);
 addDecorator(withStrictMode);
 addDecorator(withKeytipLayer);
 
+// This API is deprecated (in favor of `export const parameters = {}`) but same issue as above
 addParameters({
-  a11y: {
+  a11y: /** @type {import('@storybook/addon-a11y').A11yParameters} */ ({
     manual: true,
-  },
+  }),
 });
 
+// This API is deprecated and will no longer work in storybook 7 or with storyStoreV7.
+// https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated-configure
+// The selective loading functionality currently implemented by preview-loader can be more easily
+// handled by just settings `stories: [/*glob*/]` in main.js, but it's unclear if a replacement is
+// available for transforming files before loading as stories.
 configure(loadStories, module);
 
-export const parameters = {};
-
-// ================================
-//          Helpers
-// ================================
+// ==========================================================
+//   Helpers for loading examples in component story format
+// ==========================================================
 
 /**
  * @typedef {{
  *   default: { title: string };
- *   [subStoryName: string]: React.FunctionComponent | { title: string };
- * }} Story
+ *   [storyName: string]: React.FunctionComponent | { title: string };
+ * }} StoryGroup - Group of stories for a particular component (follows component story format)
  *
- * @typedef {{ [exportName: string]: React.ComponentType }} ComponentModule
+ * @typedef {{
+ *   [exportName: string]: React.ComponentType;
+ * }} ExampleModule - A loaded `*.Example.tsx` or `*.stories.tsx` file, exporting an example component
  */
 
-/** */
+/**
+ * Read the `*.Example.tsx` and/or `*.stories.tsx` files for the current package `start-storybook`
+ * is running in and generate a list of stories in component story format.
+ */
 function loadStories() {
-  /** @type {Map<string, Story>} */
+  /** @type {Map<string, StoryGroup>} */
   const stories = new Map();
 
-  // This shows some extra e2e-only stories
-  const includeE2E = new URLSearchParams(location.search).has('e2e');
-
   /** @type {__WebpackModuleApi.RequireContext[]} */
   const contexts = [
     // This will be updated by preview-loader with the actual current package name
     require.context('../src/PACKAGE_NAME', true, /\.(Example|stories)\.tsx$/),
   ];
 
+  // This will be replaced with the actual package name by the custom webpack loader ./preview-loader.js
+  // (note: for some reason the replacement doesn't work if the const is outside the function)
+  const packageNamePlaceholder = /** @type {string} */ ('PACKAGE_NAME');
```

Comment:
With this at the top of the file, the const was undefined when this function ran?? So I moved it back inside the function.

Reply:
interesting how it could be `undefined` when  it's in root scope, unless storybook is doing some weird magic ( which I wouldn't be surprised based on experience ). if we had tests for that we would now better I guess.

Reply:
Yeah I'm pretty confused what happened...when I looked at the runtime version of preview.js (at least the version shown in the debugger...so possibility of inaccurate sourcemaps??) it showed `const packageNamePlaceholder = 'react'` at the top of the file as expected, but when the function was actually running it was undefined. Running out of time to follow up on stuff but if you want to dig in further sometime, it would be interesting to turn off sourcemaps in the debugger and look at the actual runtime preview.js file content.